### PR TITLE
Add fx3834 [v100] Sponsored tile functionnality

### PR DESCRIPTION
**Pull request by lmarceau**
# Ticket: [fx3834](https://mozilla-hub.atlassian.net/browse/FXIOS-3834)
- Telemetry for impressions and clicks
- Telemetry for contextual menu clicks on top sites
- Long press context menu modified, with three types:
    - Sponsored tile
    - Pinned tile
    - Normal tile
- UI tests adjustment following contextual menu changes

### CODE REVIEW

**Comment by lmarceau** [comment]
@dnarcese 
# Request for data collection review form

1. What questions will you answer with this data?
- How many users sees the sponsored tiles
- How many users clicks on a sponsored tile.
- How many users use the contextual menu actions on top sites tiles.

2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?
This will help us understand usage patterns around top sites tiles and the new sponsored tiles

3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
There are no other alternatives.

4. Can current instrumentation answer these questions?
No

5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki.
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
The sponsored tile was shown | Category 2 - interaction data | https://mozilla-hub.atlassian.net/browse/FXIOS-3834
The sponsored tile was clicked | Category 2 - interaction data | https://mozilla-hub.atlassian.net/browse/FXIOS-3834
The contextual menu action was used | Category 2 - interaction data | https://mozilla-hub.atlassian.net/browse/FXIOS-3834

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
https://dictionary.telemetry.mozilla.org/apps/firefox_ios

7. How long will this data be collected? Choose one of the following:
This will expired in 6 months

8. What populations will you measure?
All channels, all locales, all countries

9. If this data collection is default on, what is the opt-out mechanism for users?
Default Glean SDK opt-out mechanism.

10. Please provide a general description of how you will analyze this data.
Glean

11. Where do you intend to share the results of your analysis?
Only on Glean, and with mobile teams.

12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
No third-party tools.

File: Client/metrics.yaml [hunk 1]
```diff
@@ -492,6 +492,40 @@ top_site:
...
+        the Firefox home page
+    extra_keys:
+      tile_url:
+        type: string
+        description: |
+          Url of the sponsored tile
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/FXIOS-3834
+    data_reviews:
+      - TODO
```
Comment (yoanarios) [inline Client/metrics.yaml:508]: is this TODO going to be handled in this PR or in another?

File: Client/metrics.yaml [hunk 2]
```diff
@@ -492,6 +492,40 @@ top_site:
...
+        menu option on top site
+    extra_keys:
+      type:
+        type: string
+        description: |
+          Type of contextual action
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/FXIOS-3834
+    data_reviews:
+      - TODO
```
Comment (yoanarios) [inline Client/metrics.yaml:525]: same data_review comment as above

**Comment by dnarcese** [comment]
r+
# Data Review Form

_1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?_ 

Yes, telemetry is documented at https://dictionary.telemetry.mozilla.org/apps/firefox_ios

_2) Is there a control mechanism that allows the user to turn the data collection on and off?_ 

Yes, Firefox iOS provides data controls in the app settings

_3) If the request is for permanent data collection, is there someone who will monitor the data over time?_

This should be re-reviewed w/ Ashley Thomas every 6mo

_4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?_

Type 2

_5) Is the data collection request for default-on or default-off?_

Default on

_6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?_

No

_7) Is the data collection covered by the existing Firefox privacy notice?_ 

Yes

_8) Does the data collection use a third-party collection tool?_ 

No

**Review by yoanarios — APPROVED** [review]

**Review by isabelrios — APPROVED** [review]
LGTM

**Comment by isabelrios** [comment]
@lmarceau I see this test failing (2 out of 2 on bitrise), is it working for you locally?
Captura de pantalla 2022-04-19 a las 11 43 37

This is the build log: https://addons-testing.bitrise.io/builds/f3aa3661-d2bb-44ed-87fc-530060bc2f97/testreport/c11fdab9-788a-46c9-9410-3860d1306207/testsuite/0/testcases?status=failed

**Pull request merged**